src/Engine/SpriteAnimator/SpriteAnimator.cpp

```diff
@@ -0,0 +1,70 @@
+#include "SpriteAnimator.h"
+
+SpriteAnimator::SpriteAnimator()
+    : initialized_(false)
+{}
+
+// Keeping this as an init right now because the objects may not
+int SpriteAnimator::init(SDL_Texture* texture, int rows, int cols, int on_every) {
+    //If there was no draw_order given as an argument, we will create a default
+    if (draw_order_.size() == 0) {
+        for (int i = 0; i < rows*cols; i++) {
+            draw_order_.push_back(i);
+        }
+    }
+    on_every_ = on_every;
+    rows_ = rows;
+    cols_ = cols;
+    //total_frames_ = rows * cols;
+    current_index_ = 0;
+    int w, h;
```

Comment:
Instead of creating these vars, just pass in `total_sprite_sheet_size_.w` and same for the height.